lib/livebook_web/live/hub/edit/team_component.ex

```diff
@@ -284,20 +305,54 @@ defmodule LivebookWeb.Hub.Edit.TeamComponent do
   end
 
   defp assign_dockerfile(socket) do
+    form = socket.assigns.dockerfile_form.params
     version = to_string(Application.spec(:livebook, :vsn))
     version = if version =~ "dev", do: "edge", else: version
+    identity_source = Livebook.Config.identity_source()
 
-    assign(socket, :dockerfile, """
+    dockerfile = """
     FROM livebook/livebook:#{version}
 
     COPY /path/to/my/notebooks /data
     ENV LIVEBOOK_APPS_PATH "/data"
 
     ENV LIVEBOOK_APPS_PATH_HUB_ID "#{socket.assigns.hub.id}"
     ENV LIVEBOOK_TEAMS_NAME "#{socket.assigns.hub.hub_name}"
-    ENV LIVEBOOK_TEAMS_OFFLINE_KEY "#{socket.assigns.hub.org_public_key}"
+    ENV LIVEBOOK_TEAMS_OFFLINE_KEY "#{socket.assigns.hub.org_public_key}"\
+    """
+
+    dockerfile =
+      if form["include_secrets"] == "true" do
+        secrets = ~s(\nENV LIVEBOOK_TEAMS_SECRETS "#{encrypt_secrets_to_dockerfile(socket)}")
+        dockerfile <> secrets
+      else
+        dockerfile
+      end
+
+    dockerfile =
+      if identity_source != :session do
+        {_module, zta} = Livebook.Config.identity_provider()
+        identity_provider = ~s(\nENV LIVEBOOK_IDENTITY_PROVIDER "#{identity_source}:#{zta}")
+
+        dockerfile <> identity_provider
+      else
+        dockerfile
+      end
+
+    cmd = ~s(\n\nCMD [ "/app/bin/livebook", "start" ])
+    assign(socket, :dockerfile, dockerfile <> cmd)
+  end
+
+  defp encrypt_secrets_to_dockerfile(socket) do
+    {secret_key, sign_secret} = Livebook.Teams.derive_keys(socket.assigns.hub.teams_key)
+
+    secrets_map =
+      for %{name: name, value: value} <- socket.assigns.secrets,
+          into: %{},
+          do: {name, Livebook.Teams.encrypt_secret_value(value, secret_key, sign_secret)}
```

Comment:
We are encrypting the value here, but we don't decrypt it while loading. We encrypt the whole map, so that should be enough, right?
```suggestion
          do: {name, value}
```